### PR TITLE
fix(cli): confine auth email content_path to project root (CLI-2320)

### PR DESCRIPTION
**Pull request by Coly010**
## Summary

`config push` resolves `auth.email.template.*.content_path` and
`auth.email.notification.*.content_path` from disk before uploading their
bytes as part of the auth config write. Resolution previously joined a
relative path onto the project root (or passed an absolute path through
unchanged) with no containment check, so a crafted `content_path` could
resolve outside the project. `content_path` is now confined to the project
root: any resolved path — a relative `..` escape or an absolute path
elsewhere on disk — aborts before the file is read, using the existing
`Invalid config for auth.email.<kind>.<name>.content_path: <reason>` error
shape.

Linear: CLI-2320 (surfaced by CLI-2313's security review; intentionally kept
out of that PR).

## What changed

- `push.auth-email-content.ts` — new `resolveContainedContentPath` normalizes
  the candidate path with `path.resolve` and rejects anything outside the
  discovered project root before `readTemplateContent` runs;
  `legacyLoadAuthEmailContent`'s template and notification loops both route
  through it. The existing notification legacy-fallback behavior
  (`supabase/`-relative lookup) is preserved as long as the resolved result
  stays inside the root.
- `push.auth-email-content.unit.test.ts` — new coverage: absolute path outside
  root (template + notification), relative `..` escape (template +
  notification), and a boundary case for a path resolving to exactly the
  project root.
- `SIDE_EFFECTS.md` — documents the containment rule in Files Read, Notes, and
  the exit-code table.

### CODE REVIEW

File: apps/cli/src/commands/config/push/push.auth-email-content.ts [hunk 1]
```diff
@@ -28,6 +31,51 @@ const EMPTY_AUTH_EMAIL_CONTENT: LegacyAuthEmailContent = {
...
+  contentPath: string,
+): string {
+  const candidate =
+    kind === "notification"
+      ? legacyResolveNotificationContentPath(cwd, contentPath)
+      : isAbsolute(contentPath)
+        ? contentPath
+        : join(cwd, contentPath);
+  const root = resolve(cwd);
+  const resolved = resolve(candidate);
```
[the remark this answers is not in the document]
Comment (Coly010) [reply apps/cli/src/commands/config/push/push.auth-email-content.ts]: Confirmed — fixed in 4ecb369ef. The containment check normalized with `path.resolve`, which is purely lexical and never follows symlinks, so an in-root symlink pointing at an outside target passed containment and `readFileSync` then followed it. Both the project root and the candidate are now resolved with `realpathSync` before the comparison (falling back to lexical resolution only when the target doesn't exist yet — no symlink to dereference in that case, left for the normal missing-file error). Added regression tests for an in-root symlink escaping to an outside target, for both a template and an `enabled: true` notification entry.

File: apps/cli/src/commands/config/push/push.auth-email-content.ts [hunk 2]
```diff
@@ -28,6 +31,51 @@ const EMPTY_AUTH_EMAIL_CONTENT: LegacyAuthEmailContent = {
...
+  if (!isPathContainedInRoot(root, resolved)) {
+    throw new Error(
+      `Invalid config for auth.email.${kind}.${name}.content_path: resolves outside the project root`,
+    );
+  }
+  return resolved;
```
[the remark this answers is not in the document]
Comment (Coly010) [reply apps/cli/src/commands/config/push/push.auth-email-content.ts]: Confirmed — same finding as the duplicate thread above, fixed in 4ecb369ef via `realpathSync` on both the project root and the candidate before the containment comparison. Regression tests added for a symlinked template and notification entry.

File: apps/cli/src/commands/config/push/push.auth-email-content.ts [hunk 3]
```diff
@@ -28,6 +31,51 @@ const EMPTY_AUTH_EMAIL_CONTENT: LegacyAuthEmailContent = {
...
 };
 
+/**
+ * Whether `candidatePath` resolves inside (or exactly to) `root`. Both
+ * arguments must already be normalized absolute paths (see `resolve`).
+ */
+function isPathContainedInRoot(root: string, candidatePath: string): boolean {
+  const rel = relative(root, candidatePath);
+  return rel === "" || (!rel.startsWith("..") && !isAbsolute(rel));
+}
```
[the remark this answers is not in the document]
Comment (Coly010) [reply apps/cli/src/commands/config/push/push.auth-email-content.ts:58]: Confirmed — fixed in 4ecb369ef. `isPathContainedInRoot` now only rejects when `rel === ".."` or `rel.startsWith(`..${sep}`)`, matching the existing correct pattern in `apps/cli/src/shared/functions/deploy.ts`'s `isContainedPath` — a same-level sibling like `..templates/invite.html` is no longer wrongly rejected.

File: apps/cli/src/commands/config/push/push.auth-email-content.ts [hunk 4]
```diff
@@ -28,6 +31,51 @@ const EMPTY_AUTH_EMAIL_CONTENT: LegacyAuthEmailContent = {
...
+  name: string,
+  cwd: string,
+  contentPath: string,
+): string {
+  const candidate =
+    kind === "notification"
+      ? legacyResolveNotificationContentPath(cwd, contentPath)
+      : isAbsolute(contentPath)
+        ? contentPath
+        : join(cwd, contentPath);
```
[the remark this answers is not in the document]
Comment (Coly010) [reply apps/cli/src/commands/config/push/push.auth-email-content.ts:88]: Agreed this asymmetry exists, but deliberately not resolving it in this PR. CLI-2320's threat model is upload: a hostile config.toml pointing content_path outside the project root gets those bytes read and uploaded to whatever project the config names — arbitrary local file exfiltration to a possibly attacker-controlled project. legacy-config-validate.ts's shared resolver and start's Kong mount are local-only (mounted into, or read display-side for, the user's own local Docker stack on their own machine, using a config they already trust enough to run `supabase start` against) — a different, much lower-severity threat model that deserves its own decision rather than folding into a security-fix PR. Filed CLI-2339 to track centralizing (or explicitly documenting the asymmetry) — https://linear.app/supabase/issue/CLI-2339.

File: apps/cli/src/commands/config/push/push.auth-email-content.ts [hunk 5]
```diff
@@ -28,6 +31,51 @@ const EMPTY_AUTH_EMAIL_CONTENT: LegacyAuthEmailContent = {
...
+  const root = resolve(cwd);
+  const resolved = resolve(candidate);
+  if (!isPathContainedInRoot(root, resolved)) {
+    throw new Error(
+      `Invalid config for auth.email.${kind}.${name}.content_path: resolves outside the project root`,
+    );
```
[the remark this answers is not in the document]
Comment (Coly010) [reply apps/cli/src/commands/config/push/push.auth-email-content.ts:94]: Confirmed — fixed in 4ecb369ef. The containment error now includes the rejected resolved path: `Invalid config for auth.email.<kind>.<name>.content_path: resolves outside the project root (<resolved path>)`.

**Review by jgoux — APPROVED** [review]
